src/agent/advertising_proxy.cpp

```diff
@@ -156,7 +156,13 @@ void AdvertisingProxy::AdvertisingHandler(otSrpServerServiceUpdateId aId,
 
 void AdvertisingProxy::PublishServiceHandler(const char *aName, const char *aType, otbrError aError, void *aContext)
 {
-    static_cast<AdvertisingProxy *>(aContext)->PublishServiceHandler(aName, aType, aError);
+    std::string       name(aName);
+    std::string       type(aType);
+    AdvertisingProxy *thisPtr = static_cast<AdvertisingProxy *>(aContext);
+
+    thisPtr->mNcp.PostTimerTask(Milliseconds(0), [name, type, aError, thisPtr]() {
```

Comment:
Should we add a `TaskRunner` to `AdvertisingProxy` rather than using `mNcp` ?

@wgtdkp comments?

Reply:
Agree.

Reply:
Added a new `TaskRunner`.